python/nav/django/settings.py

```diff
@@ -262,3 +264,40 @@
         from local_settings import *
     except ImportError:
         pass
+
+
+OIDC_AUTH = {
+    # Specify OpenID Connect endpoint. Configuration will be
+    # automatically done based on the discovery document found
+    # at <endpoint>/.well-known/openid-configuration
+    'OIDC_ENDPOINT': os.environ.get("OIDC_ENDPOINT"),
+    # The Claims Options can now be defined by a static string.
+    # ref:
+    # https://docs.authlib.org/en/latest/jose/jwt.html#jwt-payload-claims-validation
+    # The old OIDC_AUDIENCES option is removed in favor of this new option.
+    # `aud` is only required, when you set it as an essential claim.
+    ##### For NAV this is only used to specify the 'aud' claim
+    'OIDC_CLAIMS_OPTIONS': {
+        'aud': {
+            'value': os.environ.get("JWT_AUDIENCE"),
+            'essential': True,
+        }
+    },
+    # (Optional) Function that resolves id_token into user.
+    # This function receives a request and an id_token dict and expects to
+    # return a User object. The default implementation tries to find the user
+    # based on username (natural key) taken from the 'sub'-claim of the
+    # id_token.
+    'OIDC_RESOLVE_USER_FUNCTION': 'nav.web.api.v1.auth.get_user',
+    # (Optional) Time before signing keys will be refreshed (default 24 hrs)
+    'OIDC_JWKS_EXPIRATION_TIME': 24 * 60 * 60,
+    # (Optional) Time before bearer token validity is verified again
+    # (default 10 minutes)
+    'OIDC_BEARER_TOKEN_EXPIRATION_TIME': 10 * 60,
+    # (Optional) Token prefix in JWT authorization header (default 'JWT')
+    'JWT_AUTH_HEADER_PREFIX': 'Bearer',
+    # (Optional) Which Django cache to use
+    'OIDC_CACHE_NAME': 'default',
+    # (Optional) A cache key prefix when storing and retrieving cached values
+    'OIDC_CACHE_PREFIX': 'oidc_auth.',
+}
```

Comment:
Any settings we want the end-user to be able to manipulate must go in a NAV config file - again, because of the legacy status of NAV, everything is meant to be configurable through various config files and the end user is not expected to manipulate Python code files to change settings.

Typical ini file for things that affect the NAV web application (and API) would be `etc/webfront/webfront.conf`.

Reply:
> What is not supported with this is multiple simultaneous signing keys, from different parties. Is that even a thing?

That's what I was asking for in my review. Because this might be a thing we need. I.e. any VK user may want to create JWT tokens for their own purposes, using NAV - while they may simultaneously want to authorize tokens created by us/eduCSC/FEIDE.

Reply:
most of these are default values for the library itself, so they can just be removed

Reply:
The vital setting here is 
`'OIDC_RESOLVE_USER_FUNCTION': 'nav.web.api.v1.auth.get_user',`. The default is `oidc_auth.authentication.get_user_by_id`  which drags in `django.contrib.auth`, which drags in `django.contrib.contenttypes'`.

Reply:
If we drag in `django.contrib.auth` we should seriously consider also adapting NAV's "Account" to it.

Reply:
This is much cleaner now